src/main/java/team/asd/service/ProductService.java

```diff
@@ -0,0 +1,96 @@
+package team.asd.service;
+
+import lombok.NonNull;
+import org.apache.commons.collections4.CollectionUtils;
+import team.asd.constants.ProductState;
+import team.asd.entities.IsProduct;
+import team.asd.exceptions.WrongProductException;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static team.asd.constants.ProductState.Created;
+
+public class ProductService implements IsProductService {
+
+	private void checkProducts(List<IsProduct> list) throws WrongProductException {
+		if (list.stream()
+				.anyMatch(Objects::isNull)) {
+			throw new WrongProductException("Product cannot be null");
+		}
+	}
+
+	private long countProductByState(List<IsProduct> productList, ProductState state) {
+		if (CollectionUtils.isEmpty(productList)) {
+			return 0L;
+		}
+
+		return productList.stream()
+				.filter(product -> product.getState()
+						.equals(state))
+				.count();
+	}
+
+	@Override
+	public @NonNull List<String> defineProductNames(List<IsProduct> productList) throws WrongProductException {
+		if (CollectionUtils.isEmpty(productList)) {
+			return Collections.emptyList();
+		}
+		checkProducts(productList);
+		return productList.stream()
+				.map(IsProduct::getName)
+				.collect(Collectors.toList());
+	}
+
+	@Override
+	public List<IsProduct> defineProductsWithCreatedState(List<IsProduct> productList) {
+		if (CollectionUtils.isEmpty(productList)) {
+			return Collections.emptyList();
+		}
+
+		return productList.stream()
+				.filter(Objects::nonNull)
+				.filter(element -> element.getState()
+						.equals(Created))
```

Comment:
please do not use static imports for enums
https://carlosbecker.com/posts/avoid-static-imports/

Reply:
`Created.equals(element.getState())`